File: evidence/generated-systems/golang/inventory-on-golang/inventory-on-golang/internal/model/Warehouse.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Warehouse Declaration
//==============================================================
type Warehouse struct {
    gorm.Model
     Name                                    string
    Code                                    string
    Address                                                            string
    TimeZone                                    string
    AllowsOverAllocation                                    bool
     StorageLocations           []StorageLocation `gorm:"foreignKey:StorageLocationsFromWarehouseId"`
     InventoryItems           []InventoryItem `gorm:"foreignKey:InventoryItemsFromWarehouseId"`
     InboundShipments           []InboundShipment `gorm:"foreignKey:InboundShipmentsFromWarehouseId"`
     OutboundAllocations           []OutboundAllocation `gorm:"foreignKey:OutboundAllocationsFromWarehouseId"`
     OriginTransfers           []TransferOrder `gorm:"foreignKey:OriginTransfersFromWarehouseId"`
     DestinationTransfers           []TransferOrder `gorm:"foreignKey:DestinationTransfersFromWarehouseId"`
     CycleCounts           []CycleCount `gorm:"foreignKey:CycleCountsFromWarehouseId"`

// parent associations as their child

}

File: evidence/generated-systems/angular/inventory-on-angular/src/app/models/Warehouse.js

// Define collection and schema for Warehouse
export interface Warehouse {
    name:
	type : string
    code:
	type : string
    address:
	type : Address
    timeZone:
	type : string
    allowsOverAllocation:
	type : boolean
    StorageLocations:
 	type : [{ type: Schema.Types.ObjectId, ref: 'StorageLocation' }]
    InventoryItems:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InventoryItem' }]
    InboundShipments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InboundShipment' }]
    OutboundAllocations:
 	type : [{ type: Schema.Types.ObjectId, ref: 'OutboundAllocation' }]
    OriginTransfers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TransferOrder' }]
    DestinationTransfers:
 	type : [{ type: Schema.Types.ObjectId, ref: 'TransferOrder' }]
    CycleCounts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'CycleCount' }]
#
    collection: 'warehouses'
}
